tests/wakunode_rest/test_rest_admin.nim

```diff
@@ -0,0 +1,79 @@
+{.used.}
+
+import
+  std/sequtils,
+  stew/shims/net,
+  testutils/unittests,
+  presto, presto/client as presto_client,
+  libp2p/crypto/crypto
+
+import
+  ../../waku/waku_core,
+  ../../waku/waku_node,
+  ../../waku/node/peer_manager,
+  ../../waku/waku_api/rest/server,
+  ../../waku/waku_api/rest/client,
+  ../../waku/waku_api/rest/responses,
+  ../../waku/waku_api/rest/admin/types,
+  ../../waku/waku_api/rest/admin/handlers as admin_api,
+  ../../waku/waku_api/rest/admin/client as admin_api_client,
+  ../../waku/waku_relay,
+  ../testlib/wakucore,
+  ../testlib/wakunode,
+  ../testlib/testasync
+
+suite "Waku v2 Rest API - Admin":
+  var node1 {.threadvar.}: WakuNode
+  var node2 {.threadvar.}: WakuNode
+  var node3 {.threadvar.}: WakuNode
+  var peerInfo2 {.threadvar.}: RemotePeerInfo
+  var peerInfo3 {.threadvar.}: RemotePeerInfo
+  var restServer {.threadvar.}: RestServerRef
+  var client{.threadvar.}: RestClientRef
+
+  asyncSetup:
+    node1 = newTestWakuNode(generateSecp256k1Key(), ValidIpAddress.init("127.0.0.1"), Port(60600))
+    node2 = newTestWakuNode(generateSecp256k1Key(), ValidIpAddress.init("127.0.0.1"), Port(60602))
+    peerInfo2 = node2.switch.peerInfo
+    node3 = newTestWakuNode(generateSecp256k1Key(), ValidIpAddress.init("127.0.0.1"), Port(60604))
+    peerInfo3 = node3.switch.peerInfo
+
+    await allFutures(node1.start(), node2.start(), node3.start())
+    await allFutures(node1.mountRelay(), node2.mountRelay(), node3.mountRelay())
+
+    let restPort = Port(58011)
+    let restAddress = ValidIpAddress.init("127.0.0.1")
+    restServer = RestServerRef.init(restAddress, restPort).tryGet()
+
+    installAdminApiHandlers(restServer.router, node1)
+
+    restServer.start()
+
+    client = newRestHttpClient(initTAddress(restAddress, restPort))
```

Comment:
How does that work exactly? Is it call before each tests or once before.

Reply:
Exactly, it is before/after each test.